Crash in ReaderPostListActivity because of missing BottomNavController

In current `develop` branch.

```
FATAL EXCEPTION: main
                  Process: org.wordpress.android, PID: 14630
                  java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.reader.ReaderPostListActivity}: java.lang.ClassCastException: org.wordpress.android.ui.reader.ReaderPostListActivity cannot be cast to org.wordpress.android.ui.main.BottomNavController
                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
                      at android.app.ActivityThread.-wrap12(ActivityThread.java)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
                      at android.os.Handler.dispatchMessage(Handler.java:102)
                      at android.os.Looper.loop(Looper.java:154)
                      at android.app.ActivityThread.main(ActivityThread.java:6077)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
                   Caused by: java.lang.ClassCastException: org.wordpress.android.ui.reader.ReaderPostListActivity cannot be cast to org.wordpress.android.ui.main.BottomNavController
                      at org.wordpress.android.ui.reader.ReaderPostListFragment.onAttach(ReaderPostListFragment.java:340)
                      at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:937)
                      at android.app.BackStackRecord.setLastIn(BackStackRecord.java:857)
                      at android.app.BackStackRecord.calculateFragments(BackStackRecord.java:897)
                      at android.app.BackStackRecord.run(BackStackRecord.java:727)
                      at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1578)
                      at android.app.FragmentController.execPendingActions(FragmentController.java:371)
                      at android.app.Activity.performStart(Activity.java:6680)
                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2609)
```

cc @nbradbury

--- a/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java
@@ -336,8 +336,11 @@ public class ReaderPostListFragment extends Fragment
     public void onAttach(Context context) {
         super.onAttach(context);
 
-        // this will throw if parent activity doesn't implement the bottomnav controller interface
-        mBottonNavController = (BottomNavController) context;
+        // detect the bottom nav controller when this fragment is hosted in the main activity - this is used to
+        // hide the bottom nav when the user searches from the reader
+        if (context instanceof BottomNavController) {
+            mBottonNavController = (BottomNavController) context;
+        }
     }
 
     @Override

